response_operations_ui/static/js/data-panel.js

```diff
@@ -1,30 +1,32 @@
-function initDataPanels() {
-    var acc = document.getElementsByClassName("data-panel-header");
-    for (var i = 0; i < acc.length; i++) {
-        // Introduce button and hide data panel content if js running
+(function(window) {
+    function initDataPanels() {
+        const acc = document.getElementsByClassName('data-panel-header');
+        for (let i = 0; i < acc.length; i++) {
+            const initPanel = acc[i].nextElementSibling;
+            initPanel.style.display = 'none';
 
-        var initPanel = acc[i].nextElementSibling;
-        initPanel.style.display = "none";
-
-        // Add event listener to toggle displaying the data panel
-        acc[i].addEventListener("click", function() {
-            var panel = this.nextElementSibling;
-            this.classList.toggle("active");
-            if (panel.style.display === "block") {
-                panel.style.display = "none";
-            } else {
-                panel.style.display = "block";
-            }
-        });
+            // Add event listener to toggle displaying the data panel
+            acc[i].addEventListener('click', function() {
+                const panel = this.nextElementSibling;
+                this.classList.toggle('active');
+                if (panel.style.display === 'block') {
+                    panel.style.display = 'none';
+                } else {
+                    panel.style.display = 'block';
+                }
+            });
+        }
+    }
 
+    function openPanel() {
+        const panelName = window.location.hash;
+        if (panelName) {
+            document.querySelectorAll('.data-panel-header').classList.addClass('active');
+            document.querySelectorAll('.data-panel-body').style.display = 'block';
```

Comment:
Wouldn't it be best practice to 'program to interfaces, not implementations?'. Browser environments, whilst they conform better to modern day standards are still seen as volatile environments.

It doesn't have to be jQuery, it could even be a custom implementation for obtaining referenced to DOM instances/collections for something more lightweight (remember you talking about sizzle previously).

Reply:
Yeah, ideally I would move to some wrapper for `querySelector` with fallback, but to be honest, this stuff was a quick win I tacked on.

Happy if I resolve this by creating a future task to address this - bearing in mind that all this JS is non-vital?

Reply:
Agreed, there are various tickets to address this.  A lot of JS functionality on here can actually be removed, but it's a varied iterative process.

Cheers.